core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -467,9 +468,14 @@ protected void doXContentAnalyzers(XContentBuilder builder, boolean includeDefau
             if (includeDefaults) {
                 builder.field("analyzer", "default");
             }
-        } else if (includeDefaults || fieldType().indexAnalyzer().name().startsWith("_") == false && fieldType().indexAnalyzer().name().equals("default") == false) {
-            builder.field("analyzer", fieldType().indexAnalyzer().name());
-            if (fieldType().searchAnalyzer().name().equals(fieldType().indexAnalyzer().name()) == false) {
+        } else {
+            if (includeDefaults || fieldType().indexAnalyzer().name().startsWith("_") == false
+                    && fieldType().indexAnalyzer().name().equals("default") == false) {
+                builder.field("analyzer", fieldType().indexAnalyzer().name());
+            }
+            if (includeDefaults || fieldType().searchAnalyzer().name().startsWith("_") == false
+                    && fieldType().searchAnalyzer().name().equals("default") == false
```

Comment:
Instead of duplicating this complicated logic twice, can you move the search analyzer check inside (ie we would only write search_analyzer if we at least wrote analyzer)?